pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/activityinstancecreation/ActivityInstanceCreatorFromAnswers.java

```diff
@@ -0,0 +1,165 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Create  activity instances from the selected answers of parent instances.
+ * Question stable_id which answers to check is specified by parameter {@link #sourceQuestionStableId}.
+ *
+ * <p><b>Algorithm:</b>
+ * <ul>
+ *     <li>find answer (picklist) by 'sourceQuestionStableId' and detect selected options;</li>
+ *     <li>for each of selected options:</li>
+ *     <li> - create a new instance (of activity with ID='studyActivityId';</li>
+ *     <li> - if `targetQuestionStableId` is specified then copy source answers to target answers;</li>
+ * </ul>
+ */
+public class ActivityInstanceCreatorFromAnswers extends ActivityInstanceCreatorDefault {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActivityInstanceCreatorFromAnswers.class);
+
+    private final String sourceQuestionStableId;
+    private final String targetQuestionStableId;
+
+    public ActivityInstanceCreatorFromAnswers(
+            long studyActivityId,
+            String sourceQuestionStableId,
+            String targetQuestionStableId,
+            ActivityInstanceCreationService creationService) {
+        super(studyActivityId, creationService);
+        this.sourceQuestionStableId = sourceQuestionStableId;
+        this.targetQuestionStableId = targetQuestionStableId;
+    }
+
+    /**
+     * Create N activity instances.
+     * where N = number of answers in composite question `sourceQuestionStableId`.
+     * IF N > not_used_activities_count THEN N = not_used_activities_count,
+     * where not_used_activities_count = max_allowed_activity_instances - created_activity_instances.
+     */
+    @Override
+    public void create(Handle handle, EventSignal signal) {
+        JdbiActivityInstance jdbiActivityInstance = handle.attach(JdbiActivityInstance.class);
+        JdbiActivityInstanceStatus jdbiActivityInstanceStatus = handle.attach(JdbiActivityInstanceStatus.class);
+        JdbiActivity jdbiActivity = handle.attach(JdbiActivity.class);
+
+        ActivityDto activityDto = jdbiActivity.queryActivityById(studyActivityId);
+        creationService.checkSignalIfNestedTargetActivity(activityDto);
+
+        // detect parent instance id
+        Long parentInstanceId = creationService.detectParentInstanceId(activityDto);
+        if (parentInstanceId == null) {
+            throw new DDPException("Failed to detect parent instance ID");
+        }
+
+        // check if number of non-used activities is enough (it can be limited if a special limit parameter specified)
+        Integer numberOfActivitiesLeft = creationService.detectNumberOfActivitiesLeft(studyActivityId, activityDto, jdbiActivityInstance);
+        if (numberOfActivitiesLeft == null || numberOfActivitiesLeft > 0) {
+
+            List<Answer> sourceAnswers = detectSelectedAnswers(handle, parentInstanceId);
+
+            // detect number of new activities to be created
+            int newInstancesCount = sourceAnswers.size();
```

Comment:
Probably safer to turn it into a `List<SelectedPicklistOption>` so the count is more accurate?

Reply:
I looked through Pepper code and didn't found any samples of processing CompositeAnswers and producing SelectedPicklistOptions from it..
I found some examples of collecting composite child answers. Seems here the purpose is the same - to get all answers selected or entered by a user: 
```
return Optional.of(answer).stream()
                .flatMap(parent -> parent.getValue().stream())
                .flatMap(row -> row.getValues().stream())
                .filter(Objects::nonNull)
                .collect(Collectors.toList());
```
May be more proper code would be like this (but seems it's the same):
```
List<Answer> answersToCheck = new ArrayList<>();
        answersToCheck.add(answer);
        if (answer.getQuestionType() == QuestionType.COMPOSITE) {
            ((CompositeAnswer) answer).getValue().stream()
                    .map(answerRow -> answerRow.getValues())
                    .flatMap(Collection::stream)
                    .filter(each -> each != null)
                    .forEach(answersToCheck::add);
        }
```

Reply:
What I'm saying is that, right now our only use-case is having a "single-select picklist inside a composite". So it's safe to assume that "one child answer == one picklist option". But imagine if we have a "multi-select picklist inside a composite". Then, one child answer can have multiple selected options. I'm just recommending that we drill down a little more into the list of selected options for a more accurate count.

Reply:
But to copy an answer to target it is copied like this:
`new PicklistAnswer(null, targetQuestionStableId, null, picklistAnswer.getValue());`
And I believe this new answer (to be copied) contains all set of SelectedPicklistOption's (in getValue()).
So in case of multiple selected options all should be copied ok.

Reply:
Remember that the idea is, _for each_ selected option create a new instance. So if an child answer has multiple answers, I think the right thing to do is create multiples instances from that child answer. Anyways, we're not going to see that use-case come up for PanCan so I guess it's fine.

Reply:
Ah, yes! I totally forgot that each of selection in multiple selection should create a separate instance...
Ok, may be to create a separate ticket for such case? Which I could implement after PanCan